collector/pim.go

```diff
@@ -11,52 +11,47 @@ import (
 )
 
 var (
-	pimSubsystem            = "pim"
-	pimNeighborMetrixPrefix = "pim_neighbor"
-
-	pimLabels = []string{"vrf"}
-
-	pimNeighborLabels = append(pimLabels, "neighbor", "iface")
-	pimDesc           = map[string]*prometheus.Desc{
-		"neighborCount": colPromDesc(pimSubsystem, "neighbors_count_total", "Number of neighbors detected", pimLabels),
-		"upTime":        colPromDesc(pimNeighborMetrixPrefix, "uptime_seconds", "How long has the peer ben up.", pimNeighborLabels),
-	}
+	pimSubsystem = "pim"
 )
 
 func init() {
 	registerCollector(pimSubsystem, disabledByDefault, NewPIMCollector)
 }
 
 type pimCollector struct {
-	logger log.Logger
+	logger       log.Logger
+	descriptions map[string]*prometheus.Desc
 }
 
 // NewPIMCollector collects PIM metrics, implemented as per the Collector interface.
 func NewPIMCollector(logger log.Logger) (Collector, error) {
-	return &pimCollector{logger: logger}, nil
+	return &pimCollector{logger: logger, descriptions: getPIMDesc()}, nil
+}
+func getPIMDesc() map[string]*prometheus.Desc {
+	labels := []string{"vrf"}
+
+	neighborLabels := append(labels, "iface", "neighbor")
+	return map[string]*prometheus.Desc{
+		"neighborCount": colPromDesc(pimSubsystem, "neighbors_count_total", "Number of neighbors detected", labels),
+		"upTime":        colPromDesc(pimSubsystem+"_neighbor", "uptime_seconds", "How long has the peer been up.", neighborLabels),
```

Comment:
Why not
```
colPromDesc(pimSubsystem, "neighbor_uptime_seconds",[...]
```

Besides that it looks inconsistent to use neighbor and neighbors (singular and plural).

Reply:
Agreed. Fixed.